sonar-db/src/main/java/org/sonar/db/qualityprofile/ActiveRuleDao.java

```diff
@@ -55,15 +54,15 @@ public ActiveRuleDto selectOrFailByKey(DbSession session, ActiveRuleKey key) {
   }
 
   public List<ActiveRuleDto> selectByKeys(DbSession dbSession, List<ActiveRuleKey> keys) {
-    return DatabaseUtils.executeLargeInputs(keys, new KeyToDto(mapper(dbSession)));
+    return executeLargeInputs(keys, partitionOfIds -> mapper(dbSession).selectByKeys(partitionOfIds));
   }
 
   public List<ActiveRuleDto> selectByRuleId(DbSession dbSession, int ruleId) {
     return mapper(dbSession).selectByRuleId(ruleId);
   }
 
   public List<ActiveRuleDto> selectByRuleIds(DbSession dbSession, List<Integer> ids) {
-    return DatabaseUtils.executeLargeInputs(ids, new RuleIdsToDtos(mapper(dbSession)));
+    return executeLargeInputs(ids, partitionOfRuleIds -> mapper(dbSession).selectByRuleIds(partitionOfRuleIds));
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Replace this lambda with a method reference. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1612)